src/cargo/ops/cargo_add/mod.rs

```diff
@@ -611,7 +617,20 @@ fn infer_package(mut packages: Vec<Package>, src: &dyn std::fmt::Display) -> Car
                 .map(|p| p.name().as_str().to_owned())
                 .collect();
             names.sort_unstable();
-            anyhow::bail!("multiple packages found at `{src}`: {}", names.join(", "));
+            anyhow::bail!(
+                "multiple packages found at `{src}`:\n    {}{}",
+                names
+                    .iter()
+                    .map(|s| s.to_string())
+                    .coalesce(|x, y| if x.len() + y.len() < 78 {
+                        Ok(format!("{x}, {y}"))
+                    } else {
+                        Err((x, y))
+                    })
+                    .into_iter()
+                    .format("\n    "),
+                format!("\nTo disambiguate, run `cargo add --git {src} <crate>`")
```

Comment:
Do we need to format within format unnecessary?

Reply:
Good catch! Removed.